File: update_logo.js
const fs = require('fs');
const path = require('path');

const dir = './';

const navbarLogoRegex = /<a\s+href="index\.html"\s+class="logo">Genixovate\.<\/a>/g;
const navbarLogoNew = '<a href="index.html" class="logo" style="display: flex; align-items: center; text-decoration: none;"><img src="assets/logo.png" alt="Genixovate" style="height: 36px; width: auto;"></a>';

const footerLogoRegex = /<span\s+class="logo"[^>]*>Genixovate\.<\/span>/g;
const footerLogoNew = '<img src="assets/logo.png" alt="Genixovate Logo" class="logo" style="height: 48px; width: auto; margin-bottom: 20px; display: block; filter: brightness(0) invert(1);">';

const files = fs.readdirSync(dir);
const htmlFiles = files.filter(f => f.endsWith('.html'));

htmlFiles.forEach(file => {
    let content = fs.readFileSync(path.join(dir, file), 'utf8');

    content = content.replace(navbarLogoRegex, navbarLogoNew);
    content = content.replace(footerLogoRegex, footerLogoNew);

    if (!content.includes('rel="icon"')) {
        content = content.replace('</head>', '    <link rel="icon" type="image/png" href="assets/favicon.png">\n</head>');
    }

    fs.writeFileSync(path.join(dir, file), content, 'utf8');
});

console.log(`Updated ${htmlFiles.length} HTML files.`);

const jsFiles = ['update_footers.js', 'fix_how_we_build.js', 'server.js', 'update_nav.js', 'apply_seo.js', 'apply_seo_v2.js'];

jsFiles.forEach(file => {
    if (fs.existsSync(path.join(dir, file))) {
        let content = fs.readFileSync(path.join(dir, file), 'utf8');
        content = content.replace(navbarLogoRegex, navbarLogoNew);
        content = content.replace(footerLogoRegex, footerLogoNew);
        fs.writeFileSync(path.join(dir, file), content, 'utf8');
    }
});
console.log(`Updated JS files.`);

File: update_logo.py
import os
import glob
import re

html_files = glob.glob('*.html')

navbar_logo_old = r'<a\s+href="index\.html"\s+class="logo">Genixovate\.</a>'
navbar_logo_new = r'<a href="index.html" class="logo" style="display: flex; align-items: center; text-decoration: none;"><img src="assets/logo.png" alt="Genixovate" style="height: 36px; width: auto;"></a>'

footer_logo_old = r'<span\s+class="logo"[^>]*>Genixovate\.</span>'
footer_logo_new = r'<img src="assets/logo.png" alt="Genixovate Logo" class="logo" style="height: 48px; width: auto; margin-bottom: 20px; display: block; filter: brightness(0) invert(1);">'

for file_path in html_files:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Update navbar
    content = re.sub(navbar_logo_old, navbar_logo_new, content)
    
    # Update footer
    content = re.sub(footer_logo_old, footer_logo_new, content)

    # Add favicon if not present
    if 'rel="icon"' not in content:
        content = content.replace("</head>", "    <link rel=\"icon\" type=\"image/png\" href=\"assets/favicon.png\">\n</head>")
        
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
        
print(f"Updated {len(html_files)} HTML files.")

js_files = glob.glob('*.js')

for file_path in js_files:
    if os.path.exists(file_path):
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
            
        content = re.sub(navbar_logo_old, navbar_logo_new, content)
        content = re.sub(footer_logo_old, footer_logo_new, content)
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)

print(f"Updated {len(js_files)} JS files.")
